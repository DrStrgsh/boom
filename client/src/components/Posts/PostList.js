import React, { Component } from 'react';
import '../../styles/Posts.css';
import PostForPostList from './PostForPostList';

class PostList extends Component {
  render() {
      if (this.props.fetching) {
        return <p>Loading...</p>
      }
      if (this.props.error) {
        return <p>Errored, {this.props.error}</p>
      }
      
      return (
        <div className="posts-field">
          <h1 className="posts-head">Posts</h1>
          <PostForPostList posts={this.props.posts} />
        </div>
    )
  }
}

export default PostList;
