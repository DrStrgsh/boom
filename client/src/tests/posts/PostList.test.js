import React from 'react';
import PostList from '../../components/Posts/PostList';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('Post list', () => {
  const props = {
    posts: [],
    fetching: false,
    error: null
  }

  describe('Post list fetching', () => {
    const nextProps = {
      ...props,
        fetching: true
    }
    const postList = shallow(<PostList {...nextProps} />)
    
    it('fetching', () => {
      expect(postList.find('p').text()).toEqual('Loading...')
    })

    it('fetching snapshot', () => {
      expect(postList).toMatchSnapshot()
    })
  })

  describe('Post list error', () => {
    const nextProps = {
      ...props,
        error: 'Lol'
    }
    
    it('errors', () => {
      const postList = shallow(<PostList {...nextProps } />)
      expect(postList.find('p').text()).toEqual('Errored, ' + nextProps.error)
    })

    it('error snapshot', () => {
      const postList = shallow(<PostList {...nextProps } />)
      expect(postList).toMatchSnapshot()
    })
  })

  describe('Post list data', () => {
    const nextProps = {
      ...props,
        posts: [
          {
            id: 1,
            title: 'Hello',
            body: 'Ola'
          }
        ]
    }
    const postList = shallow(<PostList {...nextProps}/>)
    
    it('posts data', () => {
      expect(postList.find('h5').text()).toEqual('Hello')
    })

    it('data snapshot', () => {
      expect(postList).toMatchSnapshot()
    })
  })
})