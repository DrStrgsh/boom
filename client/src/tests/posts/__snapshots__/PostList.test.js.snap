// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post list Post list data data snapshot 1`] = `ShallowWrapper {}`;

exports[`Post list Post list error error snapshot 1`] = `ShallowWrapper {}`;

exports[`Post list Post list fetching fetching snapshot 1`] = `ShallowWrapper {}`;
